src/ecs/resources/input.rs

```diff
@@ -433,82 +390,118 @@ impl InputHandler {
+    pub fn actions_down<T: AsRef<str>>(&self, actions: &[T]) -> Result<bool, Vec<String>> {
+        let mut all_actions_are_pressed = true;
+        let mut any_action_is_pressed_this_frame = false;
         let mut bad_values = Vec::new();
         for action in actions {
-            if let Some(button) = self.actions.get(action) {
-                if !any_button_is_pressed_this_frame && self.button_down(*button) {
-                    any_button_is_pressed_this_frame = true;
+            if let Some(buttons) = self.actions.get(action.as_ref()) {
+                if !any_action_is_pressed_this_frame {
+                    if buttons.iter().any(|&b| self.button_down(b)) {
+                        any_action_is_pressed_this_frame = true;
+                    }
                 }
-                if all_buttons_are_pressed && !self.button_is_pressed(*button) {
-                    all_buttons_are_pressed = false;
+                if all_actions_are_pressed {
+                    if buttons.iter().all(|&b| !self.button_is_pressed(b)) {
+                        all_actions_are_pressed = false;
+                    }
                 }
             } else {
-                bad_values.push(*action);
+                bad_values.push(action.as_ref().to_string());
             }
         }
         if !bad_values.is_empty() {
             Err(bad_values)
         } else {
-            Ok(all_buttons_are_pressed && any_button_is_pressed_this_frame)
+            Ok(all_actions_are_pressed && any_action_is_pressed_this_frame)
         }
     }
 
     /// Assign an axis to an ID value
     ///
     /// This will insert a new axis if no entry for this id exists.
     /// If one does exist this will replace the axis at that id and return it.
-    pub fn insert_axis(&mut self, id: i32, axis: Axis) -> Option<Axis> {
+    pub fn insert_axis(&mut self, id: String, axis: Axis) -> Option<Axis> {
         self.axes.insert(id, axis)
     }
 
     /// Removes an axis, this will return the removed axis if successful.
-    pub fn remove_axis(&mut self, id: i32) -> Option<Axis> {
-        self.axes.remove(&id)
+    pub fn remove_axis<T: AsRef<str>>(&mut self, id: T) -> Option<Axis> {
+        self.axes.remove(id.as_ref())
     }
 
     /// Returns a reference to an axis.
-    pub fn get_axis(&mut self, id: i32) -> Option<&Axis> {
-        self.axes.get(&id)
+    pub fn axis<T: AsRef<str>>(&mut self, id: T) -> Option<&Axis> {
+        self.axes.get(id.as_ref())
     }
 
-    /// Assign an action to an ID value
-    ///
-    /// This will insert a new action if no entry for this id exists.
-    /// If one does exist this will replace the action at that id and return it.
-    pub fn insert_action(&mut self, id: i32, binding: Button) -> Option<Button> {
-        self.actions.insert(id, binding)
+    /// Get's a list of all axes
+    pub fn axes(&self) -> Vec<String> {
+        self.axes.keys().map(|k| k.clone()).collect::<Vec<String>>()
```

Comment:
Cuz I don't want another opaque Iterator type for this function, and this function is called very infrequently.